src/Units/Unit.java

```diff
@@ -27,37 +24,27 @@
   public UnitModel model;
   public int x;
   public int y;
+  public int ammo;
   public int fuel;
+  public int materials;
   private int captureProgress;
   private Location captureTarget;
   public Commander CO;
   public boolean isTurnOver;
   public boolean isStunned;
   private double HP;
-  public ArrayList<Weapon> weapons;
 
   public Unit(Commander co, UnitModel um)
   {
     CO = co;
     model = um;
+    ammo = model.maxAmmo;
```

Comment:
This appears to assume that a Unit can only have one weapon with limited ammo.

Reply:
Au contraire: it abstracts ammo from all weapons into one pool.
It seems counter to the abstract design to try to manage multiple separate stores of ammo per unit.

Reply:
Uhh. OK. True. So if I make a new `MechWarrior` unit that has both an AA gun and a cannon, they will both fire the same bullets? Not sure how I feel about that.

Reply:
I agree that's a little awkward.
I'd probably propose just making one of them infinite ammo, since that's the typical approach for secondary weapons - in this case, I'd go for the AA gun (since it has precedent in the Cruiser).

Reply:
Yeah, perhaps. It's fine for now anyway, since no existing units have more than one ammo-bound weapon anyway.